python/taichi/lang/ast/ast_transformer.py

```diff
@@ -717,7 +721,15 @@ def build_Attribute(ctx, node):
             node.ptr = lambda val: append(x.parent(), index, val)
         else:
             build_stmt(ctx, node.value)
-            node.ptr = getattr(node.value.ptr, node.attr)
+            if isinstance(node.value.ptr,
+                          Expr) and not hasattr(node.value.ptr, node.attr):
+                # pylint: disable-msg=C0415
+                from taichi.lang import matrix_ops as tensor_ops
+                node.ptr = getattr(tensor_ops, node.attr)
+                setattr(node, 'call_tensor_op', True)
+                setattr(node, 'caller', node.value.ptr)
```

Comment:
Will `caller` method be overriden once we build another Attribute with the same node?

Reply:
Could you give an example? I think a node should only be built once?

Reply:
Ah looks like each "Attribute" corresponds to one "attr", plz ignore this~